plugins/git/git.plugin.zsh

```diff
@@ -80,7 +80,12 @@ alias gcp='git cherry-pick'
 alias gcs='git commit -S'
 
 alias gd='git diff'
+
+#Dont add alias if gdc (GCC D Compiler) is present
+command -v gdc >/dev/null 2>&1 || {
```

Comment:
You can use zsh's `$commands` array, e.g. `if (( $+commands[gdc] )); then …`, or `if [[ -n "${commands[gdc]}" ]]`.

That should be provided through a `safe_alias` function though (see https://github.com/robbyrussell/oh-my-zsh/issues/3682#issuecomment-79344414).